### ISSUE TITLE
ASDF serialization of polynomial models missing domain (causing incorrect evaluation)

### ISSUE THREAD

**Issue by jehturner** [astropy/astropy#9809]
When saving & reloading a polynomial model, the `domain` information gets lost (also `window`?), causing the reloaded model to produce different results from the original one.

```python
>>> import numpy as np
>>> from astropy.modeling import models, fitting
>>> import asdf

>>> x = np.arange(0, 9)

>>> model = models.Chebyshev1D(1, c0=2.35, c1=2.25, domain=(0,9))
>>> model.inverse = models.Chebyshev1D(1, c0=4.5, c1=4.5, domain=[0.1, 4.6])

>>> af = asdf.AsdfFile({"model" : model})
>>> af.write_to("testdom.asdf")

>>> newmodel = asdf.open("testdom.asdf")['model']

>>> print('Fwd before {}, after {}'.format(model.domain, newmodel.domain))
Fwd before (0, 9), after None

>>> print('Inv before {}, after {}'.format(model.inverse.domain,
>>>                                        newmodel.inverse.domain))
Inv before [0.1, 4.6], after None

>>> print('y before:', model(x))
y before: [0.1 0.6 1.1 1.6 2.1 2.6 3.1 3.6 4.1]

>>> print('y after:', newmodel(x))
y after: [ 2.35  4.6   6.85  9.1  11.35 13.6  15.85 18.1  20.35]
```

See also #9795.

Linux-3.10.0-957.27.2.el7.x86_64-x86_64-with-centos-7.6.1810-Core
Python 3.6.9 |Anaconda, Inc.| (default, Jul 30 2019, 19:07:31) 
[GCC 7.3.0]
Numpy 1.17.3
Scipy 1.3.1

**Comment by jehturner** [comment]
Would an `OrthoPolynomial1D` base class from #9795 help at all with this? Currently, I think you'll have to do `hasattr(model, 'domain')` here as well -- the alternative being something like `isinstance(model, (Chebyshev1D, Hermite1D, Legendre1D, Polynomial1D))`, which you'd then have to keep up to date. But there may be other base classes with a `domain` (and user-added models), so just having `domain` as a convention throughout the package and using `hasattr()` may be more foolproof...

**Comment by jehturner** [comment]
Sorry, was that thumbs up to `hasattr()` -- or to `OrthoPolynomial1D`?

**Comment by nden** [comment]
Fixed in #9956
